### ISSUE TITLE
Linking with JACK 0.124.1: undefined reference to jack_set_port_rename_callback

### ISSUE THREAD

**Issue by diogocp** [andrewrk/libsoundio#11]
Still having trouble with JACK. Now it's a linker error.

```
Scanning dependencies of target backend_disconnect_recover
[ 21%] Building C object CMakeFiles/backend_disconnect_recover.dir/test/backend_disconnect_recover.c.o
Linking C executable backend_disconnect_recover
libsoundio.so.1.0.0: undefined reference to `jack_set_port_rename_callback'
collect2: error: ld returned 1 exit status
```

Full build log: https://gist.github.com/diogocp/d606f66c50a649d4da5c

I suppose this is the problem:

```
$ nm -D /usr/lib/libjack.so | grep jack_set_port
000000000000b820 T jack_set_port_connect_callback
000000000000b7a0 T jack_set_port_registration_callback
```

No rename. JACK bug?

**Comment by diogocp** [comment]
With JACK2 1.9.10:

```
$ nm -D /usr/lib/libjack.so | grep jack_set_port
000000000000dc70 T jack_set_port_connect_callback
000000000000dc20 T jack_set_port_registration_callback
000000000000dcc0 T jack_set_port_rename_callback
```

**Comment by andrewrk** [comment]
Here's the API spec:

http://jackaudio.org/api/group__ClientCallbacks.html#ga0c73095c1d8ebba413de6834bd7757b9

It does say `JACK_OPTIONAL_WEAK_EXPORT` though I don't see an explanation of what that means.

However, libsoundio needs this functionality, so if jack1 doesn't support it then libsoundio doesn't support jack1. Here is the issue tracker: https://github.com/jackaudio/jack1 - I don't even see an issue mentioning this.

**Comment by diogocp** [comment]
Found a recent commit adding it: jackaudio/jack1@0ac9ef7c813f4e67ab1e9a5fb4aea515fd9f1b31. Unfortunately it hasn't been released yet. :(

**Comment by diogocp** [comment]
Fixed in jack1 0.125.0.
